src/openrct2/interface/Viewport.cpp

```diff
@@ -166,24 +185,16 @@ void viewport_create(
         viewport->flags |= VIEWPORT_FLAG_GRIDLINES;
     w->viewport = viewport;
 
-    if (flags & VIEWPORT_FOCUS_TYPE_SPRITE)
-    {
-        w->viewport_target_sprite = sprite;
-        auto* centreEntity = GetEntity(sprite);
-        if (centreEntity != nullptr)
-        {
-            centrePos = { centreEntity->x, centreEntity->y, centreEntity->z };
-        }
-        else
-        {
-            log_error("Invalid entity for viewport.");
-            return;
-        }
-    }
-    else
-    {
-        w->viewport_target_sprite = SPRITE_INDEX_NULL;
-    }
+    CoordsXYZ centrePos = focus.GetPos();
+    std::visit(
+        [&w](auto&& arg) {
+            using T = std::decay_t<decltype(arg)>;
+            if constexpr (std::is_same_v<T, Focus2::CoordinateFocus>)
+                w->viewport_target_sprite = SPRITE_INDEX_NULL;
+            else if constexpr (std::is_same_v<T, Focus2::EntityFocus>)
+                w->viewport_target_sprite = arg;
+        },
+        focus.data);
```

Comment:
```suggestion
    w->viewport_target_sprite = std::visit(
        [](auto&& arg) {
            using T = std::decay_t<decltype(arg)>;
            if constexpr (std::is_same_v<T, Focus2::CoordinateFocus>)
                return SPRITE_INDEX_NULL;
            else if constexpr (std::is_same_v<T, Focus2::EntityFocus>)
                return arg;
        },
        focus.data);
```

Reply:
Oh didn't know you could return from a visit like that.